src/main/scala/impl/group1/EvolutionAnalysis.scala

```diff
@@ -0,0 +1,274 @@
+      log.info(s"maintainedPackages: $maintainedPackagesSize")
+      log.info(s"newPackages: ${newPackages.size}")
+      // calculate the nominator (interactionWithNewPackages) Packages that exist in both versions (maintainedPackages) and interact with the new added Packages (newPackages).
+      // interaction of Packages = if a class in Package A uses Objects (classes) from Package B the Packages interact and vice versa. (similar to Coupling between Objects)
+      var interactionsWithNewPackages:Double = 0
+
+      // iterate through classes for each Package to check if there are interactions with the classes from new Packages
+      // interaction between two classes is detected on the class level
+
+      // Internal Evolution Original
+      // internal Evolution Original counts if a maintainedPackage interacts with newly added Packages (counts only once)
+      // divided by the count of maintained Packages
+      if(originalMetric){
+        for(maintainedPackage <- maintainedPackages){
+          breakable{
+          for(newPackage <- newPackages){
+            val classesFromMaintainedPackage = project.classesPerPackage(maintainedPackage)
+              val classesFromNewPackage = project.classesPerPackage(newPackage)
+              for(maintainedClass <- classesFromMaintainedPackage){
+                for(newClass <- classesFromNewPackage){
+                  maintainedClass.methods.foreach(m =>
+                    if(m.body.isDefined){
+                      // interaction between two classes
+                      if(m.body.get.instructions.mkString.contains(newClass.thisType.simpleName)){
+                        log.info(s"maintainedPackageName: $maintainedPackage")
+                        log.info(s"newPackageName: $newPackage")
+                        log.info(s"Klasseninteraktion von maintainedClass: ${maintainedClass.thisType.fqn} und newClass: ${newClass.thisType.fqn}")
+                        interactionsWithNewPackages = interactionsWithNewPackages +1
+                        // if there is an interaction between the maintained and new Package it will be counted once and the rest of the iterations
+                        // for the package can be skipped, interaction between the packages is only counted once.
+                        break
+                      }
+                    }
+                  )
+                }
+              }
+            }
+          }
+        }
+
+        log.info(s"Maintained Packages interactions with new Packages: $interactionsWithNewPackages")
+
+        if(maintainedPackagesSize!=0){
+          internalEvolution = interactionsWithNewPackages/maintainedPackages.size
+        }
+      }
+      // Internal Evolution
+      // internal Evolution is the number of Packages that exist in both versions and interact with newly added Packages
+      // divided by the count of possible interactions between maintained Packages and new Packages
```

Comment:
An dieser Stelle finde ich persönlich die Erläuterung nicht optimal. Wenn ich es jetzt richtig verstanden habe, wird hier ja die Anzahl der Interaktionen zwischen Paketen gezählt. Somit finde ich die Formulierung "internal evolution is the number of packages [...]" leicht verwirrend. 
Wenn A mit B und C interagiert (B und C sind neue Pakete), wird jede Interaktion (hier 2) gezählt und nicht die Zahl irgendwelcher Pakete (ich spreche hier natürlich erstmal nur vom Zähler). Ich hoffe man versteht, was ich meine und dass ich es richtig verstanden habe. Dies soll aber eigentlich nur so eine kleine Anmerkung sein, der Code ist meiner Ansicht nach richtig.

Reply:
Danke für den Hinweis. Ich habe den Kommentar überarbeitet. 
Ist richtig das der nicht mehr eindeutig war da ich mehrfach die Metrik angepasst habe, 
hat das wohl nicht mehr soviel Sinn gemacht.